Test/Microsoft.ApplicationInsights.Test/Shared/Extensibility/TelemetryConfigurationTest.cs

```diff
@@ -285,6 +288,242 @@ public void InstrumentationKeyCanBeSetToProgrammaticallyDefineInstrumentationKey
+            Assert.AreEqual(ikey, configuration.InstrumentationKey, "instrumentation key was not set.");
+        }
+
+        [TestMethod]
+        [TestCategory("ConnectionString")]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void VerifySetConnectionString_ThrowsNullException()
+        {
+            var configuration = new TelemetryConfiguration
+            {
+                ConnectionString = null
+            };
+        }
+
+        [TestMethod]
+        [TestCategory("ConnectionString")]
+        public void VerifySetConnectionString_SetsEndpoint()
+        {
+            var explicitEndpoint = "https://127.0.0.1/";
+            var connectionString = $"InstrumentationKey=00000000-0000-0000-0000-000000000000;IngestionEndpoint={explicitEndpoint}";
+
+            var configuration = new TelemetryConfiguration
+            {
+                ConnectionString = connectionString
+            };
+
+            Assert.AreEqual("00000000-0000-0000-0000-000000000000", configuration.InstrumentationKey);
+            Assert.AreEqual(explicitEndpoint, configuration.Endpoint.Ingestion.AbsoluteUri);
+        }
+
+        [TestMethod]
+        [TestCategory("ConnectionString")]
+        public void VerifySetConnectionString_SetsChannelDefaultEndpoint()
+        {
+            var connectionString = $"InstrumentationKey=00000000-0000-0000-0000-000000000000";
+
+            var channel = new InMemoryChannel();
+
+            var configuration = new TelemetryConfiguration
+            {
+                TelemetryChannel = channel,
+                ConnectionString = connectionString,
+            };
+
+            Assert.AreEqual("00000000-0000-0000-0000-000000000000", configuration.InstrumentationKey);
+            Assert.AreEqual("https://dc.services.visualstudio.com/", configuration.Endpoint.Ingestion.AbsoluteUri);
+            Assert.AreEqual("https://dc.services.visualstudio.com/v2/track", channel.EndpointAddress);
+        }
+
+        [TestMethod]
+        [TestCategory("ConnectionString")]
+        public void VerifySetConnectionString_SetsChannelExpliticEndpoint()
+        {
+            var explicitEndpoint = "https://127.0.0.1/";
+            var connectionString = $"InstrumentationKey=00000000-0000-0000-0000-000000000000;IngestionEndpoint={explicitEndpoint}";
+
+            var channel = new InMemoryChannel();
+
+            var configuration = new TelemetryConfiguration
+            {
+                TelemetryChannel = channel,
+                ConnectionString = connectionString,
+            };
+
+            Assert.AreEqual("00000000-0000-0000-0000-000000000000", configuration.InstrumentationKey);
+            Assert.AreEqual(explicitEndpoint, configuration.Endpoint.Ingestion.AbsoluteUri);
+            Assert.AreEqual(explicitEndpoint + "v2/track", channel.EndpointAddress);
+        }
+
+        [TestMethod]
+        [TestCategory("ConnectionString")]
+        public void E2E_DefaultScenario()
```

Comment:
minor - these tests are just validating TelemetryConfiguration class, and is no way an E2E test. (in my opinion, E2E would involve TelemetryClient instance and calling Track on it .